elements/lisk-codec/test/varint.spec.ts

```diff
@@ -0,0 +1,213 @@
+				Buffer.from('00', 'hex'),
+			);
+			expect(writeVarInt(BigInt(300), { dataType: 'uint64' })).toEqual(
+				Buffer.from('ac02', 'hex'),
+			);
+			expect(writeVarInt(BigInt(2147483647), { dataType: 'uint64' })).toEqual(
+				Buffer.from('ffffffff07', 'hex'),
+			);
+			expect(writeVarInt(BigInt(4294967295), { dataType: 'uint64' })).toEqual(
+				Buffer.from('ffffffff0f', 'hex'),
+			);
+			expect(writeVarInt(BigInt('8294967295'), { dataType: 'uint64' })).toEqual(
+				Buffer.from('ffcfacf31e', 'hex'),
+			);
+			expect(
+				writeVarInt(BigInt('18446744073709551615'), { dataType: 'uint64' }),
+			).toEqual(Buffer.from('ffffffffffffffffff01', 'hex'));
+		});
+
+		it('should encode sint32', () => {
+			expect(writeSignedVarInt(0, { dataType: 'sint32' })).toEqual(
+				Buffer.from('00', 'hex'),
+			);
+			expect(writeSignedVarInt(1300, { dataType: 'sint32' })).toEqual(
+				Buffer.from('a814', 'hex'),
+			);
+			expect(writeSignedVarInt(-1300, { dataType: 'sint32' })).toEqual(
+				Buffer.from('a714', 'hex'),
+			);
+			expect(writeSignedVarInt(2147483647, { dataType: 'sint32' })).toEqual(
+				Buffer.from('feffffff0f', 'hex'),
+			);
+			expect(writeSignedVarInt(-2147483648, { dataType: 'sint32' })).toEqual(
+				Buffer.from('ffffffff0f', 'hex'),
+			);
+		});
+
+		it('should encode sint64', () => {
+			expect(writeSignedVarInt(BigInt(0), { dataType: 'sint64' })).toEqual(
+				Buffer.from('00', 'hex'),
+			);
+			expect(writeSignedVarInt(BigInt(1300), { dataType: 'sint64' })).toEqual(
+				Buffer.from('a814', 'hex'),
+			);
+			expect(writeSignedVarInt(BigInt(-1300), { dataType: 'sint64' })).toEqual(
+				Buffer.from('a714', 'hex'),
+			);
+			expect(
+				writeSignedVarInt(BigInt(2147483647), { dataType: 'sint64' }),
+			).toEqual(Buffer.from('feffffff0f', 'hex'));
+			expect(
+				writeSignedVarInt(BigInt(-2147483648), { dataType: 'sint64' }),
+			).toEqual(Buffer.from('ffffffff0f', 'hex'));
+			expect(
+				writeSignedVarInt(BigInt('9223372036854775807'), {
+					dataType: 'sint64',
+				}),
+			).toEqual(Buffer.from('feffffffffffffffff01', 'hex'));
+			expect(
+				writeSignedVarInt(BigInt('-9223372036854775808'), {
+					dataType: 'sint64',
+				}),
+			).toEqual(Buffer.from('ffffffffffffffffff01', 'hex'));
+		});
+	});
+
+	describe('reader', () => {
+		it('should decode uint32', () => {
+			expect(
+				readVarInt(Buffer.from('00', 'hex'), { dataType: 'uint32' }),
+			).toEqual(0);
+			expect(
+				readVarInt(Buffer.from('ac02', 'hex'), { dataType: 'uint32' }),
+			).toEqual(300);
+			expect(
+				readVarInt(Buffer.from('ffffffff07', 'hex'), { dataType: 'uint32' }),
+			).toEqual(2147483647);
+			expect(
+				readVarInt(Buffer.from('ffffffff0f', 'hex'), { dataType: 'uint32' }),
+			).toEqual(4294967295);
+		});
+
+		it('should fail to decode uint32 when input is out of range', () => {
+			expect(() =>
+				readVarInt(Buffer.from('ffffffff7f', 'hex'), { dataType: 'uint32' }),
```

Comment:
this is out of range (the test is good). But if we want the smallest out of range, then we could test '8080808010'. Which will throw accordingly (the 5th byte is larger than 0f)